Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -22,6 +28,171 @@ def __init__(self, port):
+
+    for decision in policy_decisions:
+        if decision not in VALID_POLICY_DECISIONS:
+            raise InvalidMultiSelectException('policy_decisions', VALID_POLICY_DECISIONS)
+
+    if protocol not in VALID_PROTOCOLS:
+        raise InvalidSingleSelectException(protocol, "protocol", VALID_PROTOCOLS)
+
+
+def validate_arguments_for_illumio_virtual_service_create_command(name: str | None, port: str | None,
+                                                                  protocol: str) -> tuple:
+    """Validate and convert arguments for illumio virtual service create command.
+
+    Args:
+        name: Name of virtual service.
+        port: Port number.
+        protocol: Protocol name.
+
+    Returns:
+        Returns tuple of converted values (port, protocol).
+    """
+    validate_required_parameters(name=name, port=port)
+
+    port = arg_to_number(port, arg_name="port")  # type: ignore
+
+    if port != -1 and (port > MAX_PORT or port < MIN_PORT):  # type: ignore
+        raise ValueError("{} is an invalid value for port. Value must be in 1 to 65535 or -1.".format(port))
+
+    if protocol not in VALID_PROTOCOLS:
+        raise InvalidSingleSelectException(protocol, 'protocol', VALID_PROTOCOLS)
+
+    proto = convert_protocol(protocol)
+    return port, proto
+
+
+
+def prepare_hr_for_traffic_analysis(response: list, protocol: str) -> str:
+    """Prepare Human Readable output for traffic analysis command.
+
+    Args:
+        response: Response from the SDK.
+        protocol: Communication protocol.
+
+    Returns:
+        markdown string to be displayed in the war room.
+    """
+    hr_output = []
+    for traffic in response:
+        hr_output.append({
+            "Source IP": traffic.get('src', {}).get('ip'),
+            "Destination IP": traffic.get('dst', {}).get('ip'),
+            "Destination Workload Hostname": traffic.get('dst', {}).get('workload', {}).get('hostname'),
+            "Service Port": traffic.get('service', {}).get('port'),
+            "Service Protocol": protocol,
+            "Policy Decision": traffic.get('policy_decision'),
+            "State": traffic.get("state"),
+            "Flow Direction": traffic.get('flow_direction'),
+            "First Detected": arg_to_datetime(traffic.get('timestamp_range', {}).get('first_detected', '')).strftime(  # type: ignore # noqa
+                HR_DATE_FORMAT),  # type: ignore # noqa
+            "Last Detected": arg_to_datetime(traffic.get('timestamp_range', {}).get('last_detected', '')).strftime(  # type: ignore # noqa
+                HR_DATE_FORMAT)
+        })
+
+    headers = ["Source IP", "Destination IP", "Destination Workload Hostname", "Service Port", "Service Protocol",
+               "Policy Decision", "State", "Flow Direction", "First Detected", "Last Detected"]
+
+    return tableToMarkdown("Traffic Analysis:", hr_output, headers=headers, removeNull=True)
+
+
+def prepare_hr_for_illumio_virtual_service_create_command(result: dict) -> str:
```

Comment:
Similar to above, this function name can be shortened. At least remove `illumio` - it's implied by the module context. `prepare_virtual_service_output` or similar is probably good enough.

Reply:
Yes, I think it's better to be internally consistent, and in my opinion the `prepare_hr_for` prefix is unclear.

Reply:
Okay, works